plugins/modules/net_tools/ldap/ldap_search.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2016, Peter Sagerson <psagers@ignorare.net>
+# Copyright: (c) 2020, Sebastian Pfahl <eryx@gmx.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r"""
+---
+module: ldap_search
+short_description: Search for entries in a LDAP server.
+description:
+  - Return the results of an LDAP search. Use Ansible's 'register' statement.
+notes:
+  - Only supports Python3
```

Comment:
```suggestion
  - Only supports Python 3.
```

Reply:
Is this a requirement of the `ldap` Python library?

Reply:
Actually this is a requirement of the way some parts of the module work. Unfortunately I'm not the best Python programmer and was not able to get this module to work with both versions.

Reply:
Which parts do not work with Python 2?

Reply:
You're right, the first version I've coded was not working because of some iterations, but those are gone now since I use the LdapGeneric module now.